acceptance/tests/reports/finalized_on_cycle.rb

```diff
@@ -0,0 +1,34 @@
+test_name "Reports are finalized on resource cycles"
+
+check_script = <<CHECK
+require 'yaml'
+require 'puppet'
+
+exit YAML.load_file(ARGV[0]).metrics.empty? ? 1 : 0
+CHECK
+
+cyclic_manifest = <<MANIFEST
+notify { 'foo':
+  require => Notify['bar']
+}
+
+notify { 'bar':
+  require => Notify['foo']
+}
+MANIFEST
+
+agents.each do |agent|
+  tmpdir = agent.tmpdir('report_finalized')
+  ruby = "#{agent['privatebindir']}/ruby"
```

Comment:
lib/puppet/acceptance/common_utils.rb has a util for getting the local ruby, `Puppet::Acceptance::CommandUtils.ruby_command`. See tests/windows/eventlog.rb for usage.